### PR TITLE
rename jobs with cni

### PR DESCRIPTION
**Pull request by jluhrsen**
this appends -ovn or -sdn to any job with a flake complaining that
the job name does not match the underlying CNI (ovn or sdn) and
also has a name already following one of the following formats:

e2e-aws
e2e-gcp
e2e-azure
e2e-metal-ipi

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

File: ci-operator/jobs/openshift/builder/openshift-builder-master-presubmits.yaml [hunk 1]
```diff
@@ -6,15 +6,15 @@ presubmits:
     - ^master$
     - ^master-
     cluster: build03
-    context: ci/prow/e2e-aws
+    context: ci/prow/e2e-aws-builds
```
Comment (deads2k) [inline ci-operator/jobs/openshift/builder/openshift-builder-master-presubmits.yaml:9]: was this change intentional?

File: ci-operator/jobs/openshift/builder/openshift-builder-master-presubmits.yaml [hunk 2]
```diff
@@ -72,22 +72,23 @@ presubmits:
...
-    trigger: (?m)^/test( | .* )e2e-aws,?($|\s.*)
+    trigger: (?m)^/test( | .* )e2e-aws-builds,?($|\s.*)
   - agent: kubernetes
     always_run: false
     branches:
     - ^master$
     - ^master-
     cluster: build03
-    context: ci/prow/e2e-aws-builds
+    context: ci/prow/e2e-aws-builds-techpreview
```
Comment (deads2k) [inline ci-operator/jobs/openshift/builder/openshift-builder-master-presubmits.yaml:82]: what about this one?

File: ci-operator/jobs/openshift/builder/openshift-builder-master-presubmits.yaml (hunk 1 above)
Comment (jluhrsen) [reply ci-operator/jobs/openshift/builder/openshift-builder-master-presubmits.yaml:9]: yeah, this part is ugly. make jobs re-arranges things with the new names. makes the diffs here in GH hard to read.

File: ci-operator/jobs/openshift/cluster-baremetal-operator/openshift-cluster-baremetal-operator-master-presubmits.yaml [hunk 3]
```diff
@@ -243,8 +243,8 @@ presubmits:
         - --lease-server-credentials-file=/etc/boskos/credentials
         - --report-credentials-file=/etc/report/credentials
         - --secret-dir=/secrets/ci-pull-credentials
-        - --secret-dir=/usr/local/e2e-metal-ipi-cluster-profile
-        - --target=e2e-metal-ipi
+        - --secret-dir=/usr/local/e2e-metal-ipi-ovn-dualstack-cluster-profile
```
Comment (deads2k) [inline ci-operator/jobs/openshift/cluster-baremetal-operator/openshift-cluster-baremetal-operator-master-presubmits.yaml:246]: is somethign alpha sorting in the script maybe?

Comment (jluhrsen) [reply ci-operator/jobs/openshift/cluster-baremetal-operator/openshift-cluster-baremetal-operator-master-presubmits.yaml:246]: yes, I think so. I got stuck on this for a while in the beginning of my renaming journey. couldn't understand all the wacky changes that 'make jobs' was doing.

should be ok though.

**Comment by bparees** [comment]
there are a few jobs mentioned in the release controller config i'd have expected would be renamed, is there a reason they weren't?

https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L110

https://github.com/openshift/release/blob/686e4b6beacdc50509e5bfa96c7e8da7dc611f30/core-services/release-controller/_releases/release-ocp-4.12.json#L158

https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L170

https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L182

https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L237

etc.

**Comment by jluhrsen** [comment]
> there are a few jobs mentioned in the release controller config i'd have expected would be renamed, is there a reason they weren't?
> 
> https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L110
> 
> https://github.com/openshift/release/blob/686e4b6beacdc50509e5bfa96c7e8da7dc611f30/core-services/release-controller/_releases/release-ocp-4.12.json#L158
> 
> https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L170
> 
> https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L182
> 
> https://github.com/openshift/release/blob/84211dbd5bc765185d67ac98333d4cdbdf8d69e4/core-services/release-controller/_releases/release-ocp-4.12.json#L237
> 
> etc.

There are a lot of jobs that will still need to be renamed. I'm doing this activity in batches. this batch was the easier one but gets ~40 done quickly. more PRs will be coming.

**Comment by bparees** [comment]
> There are a lot of jobs that will still need to be renamed. I'm doing this activity in batches. this batch was the easier one but gets ~40 done quickly. more PRs will be coming.

got it.

/approve

**Comment by jluhrsen** [comment]
https://issues.redhat.com/browse/TRT-487
